web_tree_date_search/static/src/js/web_tree_date_search.js

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2015 Noviat nv/sa (www.noviat.com)
+
+openerp.web_tree_date_search = function(instance) {
+    var _t = instance.web._t,
+       _lt = instance.web._lt;
+    var QWeb = instance.web.qweb;
+
+    instance.web.ListView.include({
+        init: function(parent, dataset, view_id, options) {
+            this._super.apply(this, arguments);
+            if ("dates_filter" in dataset.context){
+                this.dates_filter = dataset.context["dates_filter"];
+                this.current_date_from  = [];
+                this.current_date_to  = [];
+            }
+            this.tree_date_search_loaded = false;
+        },
+        do_load_state: function(state, warm) {
+            var ui_toolbar_loc = $('.ui-toolbar:last');
+            if (this.dates_filter && this.dates_filter.length > 0){
+                ui_toolbar_loc.show();
+            }
+            else{
+                if (ui_toolbar_loc.children().length == 0)
+                    ui_toolbar_loc.hide();
+            }
+            return this._super.apply(this, arguments);
+        },
+        load_list: function(data) {
+            var self = this;
+            var tmp = this._super.apply(this, arguments);
+            if (!this.tree_date_search_loaded){
+                this.date_field_data = [];
+                for (i in this.dates_filter){
+                    var date_field = this.dates_filter[i];
+                    for (col in this.columns){
```

Comment:
Is there a way to improve this code instead of looping to all the fields?
Not a breaking point but just a thought. Would be better if it is possible.

Reply:
There are [ways to write it shorter](https://stackoverflow.com/questions/12744447/underscore-js-find-item-by-id), but it won't affect speed, since such abstractions to my knowledge just perform the same loop under the hood.